File: 6주차/ain/programmers6_2.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class programmers6_2 {

    static class Solution {
        public static int solution(int k, int[] tangerine) {
            int answer = 0;

            Map<Integer, Integer> basket = new HashMap<>();
            for(int size : tangerine) {
                //사이즈별 귤 개수 세팅
                basket.put(size, basket.getOrDefault(size, 0) + 1);
            }

            List<Integer> keySet = new ArrayList<>(basket.keySet());

            //사이즈별 귤 개수가 많은 순서대로 내림차순 정렬
            keySet.sort((o1, o2) -> basket.get(o2).compareTo(basket.get(o1)));

            int i = 0;
            while(k > 0) {
                k -= basket.get(keySet.get(i));
                answer++;
                i++;
            }

            return answer;
        }
    }

    public static void main(String[] args) {
        int k = 6;
        int[] tangerine = {1, 3, 2, 5, 4, 5, 2, 3};

        System.out.println(Solution.solution(k, tangerine));
    }
}

File: 6주차/yeonjin/programmers6_2.py
def solution(k, tangerine):
    tmp_list = []
    tangerine.sort()
    cnt = 1
    for i in range(len(tangerine)):
        if i == len(tangerine)-1 :
            tmp_list.append([tangerine[i], cnt])
        elif tangerine[i] == tangerine[i + 1]:
            cnt += 1
        else :
            tmp_list.append([tangerine[i], cnt])
            cnt = 1
    tmp_list.sort(key = lambda x:-x[1])  
    
    total = 0
    result = 0

    for info in tmp_list:
        total += info[1]
        result += 1
        if k <= total :
            break       


    answer = result
    return answer

